app/services/preserved_object_handler.rb

```diff
@@ -191,8 +191,9 @@ def confirm_version_in_catalog
     results = []
     confirm_results = with_active_record_transaction_and_rescue do
       po_db_object = PreservedObject.find_by!(druid: druid)
+      pc_db_object = PreservedCopy.find_by!(preserved_object: po_db_object, endpoint: endpoint) if po_db_object
+      return unless pc_db_object
```

Comment:
i am very slightly surprised this didn't require more edits to the tests, but i think i like this general approach better than what was there before.

however, i think you'd want to maybe append to the `results` list and return that?  or, if you just want to get out of the `with_active_record_transaction_and_rescue` block, you could call `break`.  but i'm assuming you want to return.

now that i think about it more, i'm also slightly unsure what happens when you call `return` in the middle of a transaction.  i'd hope it either implicitly aborts the transaction, or implicitly commits it, or something similarly intuitive (though i'd find the former approach much more immediately intuitive than the latter).  i'm happy to look into that if you did want to keep the `return`, as i think i'd like to know the expected behavior in that situation.

Reply:
break is a better choice; I will fix.